internal/ingress/controller/template/template.go

```diff
@@ -288,42 +289,37 @@ func buildResolvers(res interface{}, disableIpv6 interface{}) string {
 	return strings.Join(r, " ") + ";"
 }
 
-func needsRewrite(location *ingress.Location) bool {
-	if len(location.Rewrite.Target) > 0 && location.Rewrite.Target != location.Path {
-		return true
-	}
-	return false
-}
+// ByPath implements sort.Interface based on the path feild
+type ByPath []*ingress.Location
 
-// atLeastOneNeedsRewrite checks if the nginx.ingress.kubernetes.io/rewrite-target annotation is used on the '/' path
-func atLeastOneNeedsRewrite(input interface{}) bool {
+func (p ByPath) Len() int           { return len(p) }
+func (p ByPath) Less(i, j int) bool { return len(p[i].Path) < len(p[j].Path) }
+func (p ByPath) Swap(i, j int)      { p[i], p[j] = p[j], p[i] }
+
+func orderLocations(input interface{}) []*ingress.Location {
```

Comment:
We already sort the locations https://github.com/kubernetes/ingress-nginx/blob/dfa4f8a61e44cf1a92134381cd28a365fe385734/internal/ingress/controller/controller.go#L606
(at this point in the code the list should be already ordered)

Reply:
For my implementation they need to be ordered in descending length. Would wrapping this sort in `sort.Reverse` have any negative impact? I.e. is there a reason why they're already being sorted?

Reply:
> is there a reason why they're already being sorted?

Avoid unnecessarily reloads because of random order and to ensure `/` is the last location

Reply:
Oh okay great, so it is already sorting in descending order.